cg/models/balsamic/config.py

```diff
@@ -1,7 +1,8 @@
 from datetime import datetime
 from pathlib import Path
 
-from pydantic.v1 import BaseModel, validator
+from pydantic import BaseModel, field_validator
+from pydantic_core.core_schema import ValidationInfo
```

Comment:
You could simplify this import by importing ValidationInfo directly from pydantic
```suggestion
from pydantic import BaseModel, field_validator, ValidationInfo
```